### PR TITLE
1085 updating the hcl whitelist to include all supported options

### LINKED ISSUE #1085: Set session_prefix rule via consul_acl
##### SUMMARY
I can manually set in Consul:
```
session_prefix "" {
  policy = "write"
}
```

But via consul_acl I get errors.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
consul_acl

##### ADDITIONAL INFORMATION

```yaml
  - name: Create an acl with specific token with both key and service rules
    consul_acl:
      mgmt_token: "{{ consul_mgmt_token }}"
      name: Token for traefik
      token: "{{ consul_traefik_token }}"
      rules:
        - key: traefik
          policy: write
        - session: ""
          policy: write
        - session_prefix: ""
          policy: write
```

Error:
```
TASK [traefik : Create an acl with specific token with both key and service rules] *************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: A rule requires one of agent/event/key/keyring/node/operator/query/service/session and a policy.
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib/python3.8/runpy.py\", line 206, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.8/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.8/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_consul_acl_payload_r7_luy0y/ansible_consul_acl_payload.zip/ansible/modules/clustering/consul_acl.py\", line 663, in <module>\n  File \"/tmp/ansible_consul_acl_payload_r7_luy0y/ansible_consul_acl_payload.zip/ansible/modules/clustering/consul_acl.py\", line 639, in main\n  File \"/tmp/ansible_consul_acl_payload_r7_luy0y/ansible_consul_acl_payload.zip/ansible/modules/clustering/consul_acl.py\", line 426, in decode_rules_as_yml\nValueError: A rule requires one of agent/event/key/keyring/node/operator/query/service/session and a policy.\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

### PR DESCRIPTION
**Pull request by iridian-ks**
##### SUMMARY

Fixes #1085 and all possible other tickets related to adding support for all ACL options

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

consul_acl

##### ADDITIONAL INFORMATION

Full list of rules are here: https://www.consul.io/docs/security/acl/acl-rules

### CODE REVIEW

File: changelogs/fragments/1085-consul-acl-hcl-whitelist-update.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - consul_acl - update the hcl whitelist to include all supported options
```
Comment (felixfontein) [inline changelogs/fragments/1085-consul-acl-hcl-whitelist-update.yml:2]:
```suggestion
  - consul_acl - update the hcl allowlist to include all supported options (https://github.com/ansible-collections/community.general/pull/2495).
```

**Comment by felixfontein** [comment]
(You can ignore the failing third-party-check. It currently fails a lot..)

**Comment by felixfontein** [comment]
recheck

**Comment by felixfontein** [comment]
recheck

**Comment by iridian-ks** [comment]
Thank you for making the suggested fix. It's much appreciated.

**Review by felixfontein — APPROVED** [review]
I cannot test this and I don't know the code, but it looks like it should not break anything. If nobody complains before the next release, I'm going to merge this.

**Review by aminvakil — APPROVED** [review]
I don't use this module either, but change looks good to me.

**Comment by felixfontein** [comment]
@iridian-ks thanks for this fix!
@aminvakil thanks for reviewing!

**Pull request merged**